### PR TITLE
Flow integration

### PR DESCRIPTION
**Pull request by felipecalvo**
:warning: **We need to tag and release flow and domain first**

This PR integrates flow and fixes some specs (as requested in https://github.com/mumuki/mumuki-laboratory/pull/1378) which failed on the latest domain changes.

Once we have new releases for flow and domain, they should be removed from Gemfile and added to the gemspec file.

### CODE REVIEW

File: spec/dummy/db/schema.rb [hunk 1]
```diff
@@ -172,6 +175,7 @@
     t.text "choices"
     t.text "settings"
     t.text "custom_expectations"
+    t.integer "educational_purpose", default: 0
```
Comment (flbulgarelli) [inline spec/dummy/db/schema.rb]: The term educational seems redundant to me, since mostly everything in laboratory is education-oriented

Comment (luchotc) [reply spec/dummy/db/schema.rb]: I agree with that but he have a _type_ and a _goal_ field in exercise. Adding just purpose seems too vague in this context.

File: app/helpers/concerns/with_student_path_navigation.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ def next_lesson_button(guide)
   end
 
   def next_exercise_button(exercise)
-    next_button(exercise) || next_button(exercise.guide.lesson)
+    AdaptiveNavigation.new(self).button(exercise) || next_button(exercise.guide.lesson)
```
Comment (luchotc) [inline app/helpers/concerns/with_student_path_navigation.rb]: Can we delegate this to a method like `adaptive_navigation_button`?

File: spec/dummy/db/schema.rb (hunk 1 above)
Comment (flbulgarelli) [reply spec/dummy/db/schema.rb]: I disagree.  Type and goal have clear semantics.  The first is the exercise style like problem or playground. The latter is a term extensively used in mumukit, domain and runners and describes a way of evaluation - as opposed to tests and expectations.

Educational does not remove any ambiguity here.

Comment (felipecalvo) [reply spec/dummy/db/schema.rb]: The name comes from a discussion we had a couple months ago. I commented many other options we had considered on the [original commit](https://github.com/mumuki/mumuki-domain/commit/0280fff33cb3035b3dc66bf03321b843527eec1e) on Domain, maybe there's something there that suits the idea better.

Comment (flbulgarelli) [reply spec/dummy/db/schema.rb]: If we want to remove any ambiguity perhaps we should use the term flow or didactic sequence.

Or something like motivation. motivation_in_flow for instance.

File: app/helpers/concerns/with_student_path_navigation.rb (hunk 2 above)
Comment (felipecalvo) [reply app/helpers/concerns/with_student_path_navigation.rb]: Done, I think. Check it out?

**Comment by felipecalvo** [comment]
This is not happening soon and might have to be re-done anyway.

**Pull request closed without merging**